pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/CloseResource.xml

```diff
@@ -1385,6 +1385,27 @@ public class Foo {
         }
         Closeables.close(stream, false);
     }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>[java] [6.27.0] Exception applying rule CloseResource on file ... java.lang.NullPointerException #2755</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.io.*;
+
+public class Foo {
+    public int bar() {
+        var inputStream = getInputStreamFromSomewhere();
```

Comment:
I don't really like that this method is undefined. In PMD 7 we have good type resolution, but this method call, and the variable, would have type `TypeSystem.UNKNOWN`. If we want to use type resolution in this rule later to improve its precision/generality, we have to use "compilable" code in tests.

You can just define the method in the file, so that typeres can see the signature:
```
InputStream getInputStreamFromSomewhere() { return null; }
```

Note that this is not specific to this PR, but really is a common problem in our tests, because often we just copy-paste the test case provided in the issue, where many types are unresolved.